test/views/issueish-list-view.test.js

```diff
@@ -0,0 +1,145 @@
+const noStatus = new Issueish(createPullRequestResult({number: 4, states: null}));
+
+class CustomComponent extends React.Component {
+  render() {
+    return <div className="custom" />;
+  }
+}
+
+describe('IssueishListView', function() {
+  let branch, branchSet;
+
+  beforeEach(function() {
+    branch = new Branch('master', nullBranch, nullBranch, true);
+    branchSet = new BranchSet();
+    branchSet.add(branch);
+  });
+
+  function buildApp(overrideProps = {}) {
+    return (
+      <IssueishListView
+        title="aaa"
+        isLoading={true}
+        total={0}
+        issueishes={[]}
+
+        onCreatePr={() => {}}
+        onIssueishClick={() => {}}
+        onMoreClick={() => {}}
+
+        {...overrideProps}
+      />
+    );
+  }
+
+  it('sets the accordion title to the search name', function() {
+    const wrapper = shallow(buildApp({
+      title: 'the search name',
+    }));
+    assert.strictEqual(wrapper.find('Accordion').prop('leftTitle'), 'the search name');
+  });
+
+  describe('while loading', function() {
+    it('sets its accordion as isLoading', function() {
+      const wrapper = shallow(buildApp());
+      assert.isTrue(wrapper.find('Accordion').prop('isLoading'));
+    });
+
+    it('passes an empty result list', function() {
+      const wrapper = shallow(buildApp());
+      assert.lengthOf(wrapper.find('Accordion').prop('results'), 0);
+    });
+  });
+
+  describe('with empty results', function() {
+    it('uses a custom EmptyComponent if one is provided', function() {
+      const wrapper = mount(buildApp({isLoading: false, emptyComponent: CustomComponent}));
+
+      assert.isTrue(wrapper.find('CustomComponent').exists());
+    });
+
+    it('renders an error tile if an error is present', function() {
+      const error = new Error('error');
+      error.rawStack = error.stack;
+      const wrapper = mount(buildApp({isLoading: false, error}));
+
+      assert.isTrue(wrapper.find('QueryErrorTile').exists());
+    });
+  });
+
+  describe('with nonempty results', function() {
+    it('passes its results to the accordion', function() {
+      const issueishes = [allGreen, mixed, allRed];
+      const wrapper = shallow(buildApp({
+        isLoading: false,
+        total: 3,
+        issueishes,
+      }));
+      assert.deepEqual(wrapper.find('Accordion').prop('results'), issueishes);
+    });
+
+    it('renders a check if all status checks are successful', function() {
+      const wrapper = mount(buildApp({isLoading: false, total: 1, issueishes: [allGreen]}));
+      assert.isTrue(wrapper.find('span.github-IssueishList-item--status').hasClass('icon-check'));
+    });
+
+    it('renders an x if all status checks have failed', function() {
+      const wrapper = mount(buildApp({isLoading: false, total: 1, issueishes: [allRed]}));
+      assert.isTrue(wrapper.find('span.github-IssueishList-item--status').hasClass('icon-x'));
+    });
+
+    it('renders a donut chart if status checks are mixed', function() {
```

Comment:
![image](https://user-images.githubusercontent.com/3781742/41687686-78f78108-749d-11e8-9150-93eeb6e835a9.png)

Please let me know if you get tired/annoyed of the bitmoji thing, I can stop.

Reply:
> Please let me know if you get tired/annoyed of the bitmoji thing, I can stop.

I'll be sure to let you know when it stops making me laugh every time 😆